_studio/shared/src/libmfx_core_hw.cpp

```diff
@@ -0,0 +1,114 @@
+#include "libmfx_core_hw.h"
+
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <dirent.h>
+#include <string>
+#include <list>
+#include <fstream>
+#include <iomanip>
+
+const std::string PCI_DIR = "/sys/bus/pci/devices";
```

Comment:
PCI_DIR is definitely Linux specifc, Class ID and Controller Class - not sure, but quite possible that on Windows there will be absolutely different method to get this info.